diff --git chrome/chrome_paks.gni chrome/chrome_paks.gni
--- chrome/chrome_paks.gni
+++ chrome/chrome_paks.gni
@@ -288,7 +288,7 @@ template("chrome_extra_paks") {
         "//extensions:extensions_resources",
       ]
     }
-    if (enable_plugins) {
+    if (enable_plugins && enable_pdf) {
       sources += [ "$root_gen_dir/chrome/pdf_resources.pak" ]
       deps += [ "//chrome/browser/resources/pdf:resources" ]
     }
