diff --git chrome/chrome_paks.gni chrome/chrome_paks.gni
--- chrome/chrome_paks.gni
+++ chrome/chrome_paks.gni
@@ -4,6 +4,7 @@
 
 import("//build/config/chromeos/ui_mode.gni")
 import("//build/config/locales.gni")
 import("//cef/libcef/features/features.gni")
+import("//pdf/features.gni")
 import("//chrome/browser/buildflags.gni")
 import("//chrome/browser/resources/pdf/ink/ink.gni")
@@ -292,7 +293,7 @@ template("chrome_extra_paks") {
         "//extensions:extensions_resources",
       ]
     }
-    if (enable_plugins) {
+    if (enable_plugins && enable_pdf) {
       sources += [ "$root_gen_dir/chrome/pdf_resources.pak" ]
       deps += [ "//chrome/browser/resources/pdf:resources" ]
     }
