diff --git chrome/common/printing/printer_capabilities.cc chrome/common/printing/printer_capabilities.cc
index 45eefbbf0..73622a8fc 100644
--- chrome/common/printing/printer_capabilities.cc
+++ chrome/common/printing/printer_capabilities.cc
@@ -29,4 +29,5 @@
 #include "base/strings/string_util.h"
 #include "base/strings/utf_string_conversions.h"
 #include "components/strings/grit/components_strings.h"
+#include "printing/buildflags/buildflags.h"
 #include "ui/base/l10n/l10n_util.h"
@@ -143,8 +144,12 @@ std::string GetUserFriendlyName(const std::string& printer_name) {
       printer_name_trimmed, "\\", base::KEEP_WHITESPACE, base::SPLIT_WANT_ALL);
   if (tokens.size() != 2 || tokens[0].empty() || tokens[1].empty())
     return printer_name;
+#if BUILDFLAG(ENABLE_PRINT_PREVIEW)
   return l10n_util::GetStringFUTF8(
       IDS_PRINT_PREVIEW_FRIENDLY_WIN_NETWORK_PRINTER_NAME,
       base::UTF8ToUTF16(tokens[1]), base::UTF8ToUTF16(tokens[0]));
+#else
+  return printer_name;
+#endif
 }
 #endif
