diff --git chrome/browser/extensions/api/resources_private/resources_private_api.cc chrome/browser/extensions/api/resources_private/resources_private_api.cc
--- chrome/browser/extensions/api/resources_private/resources_private_api.cc
+++ chrome/browser/extensions/api/resources_private/resources_private_api.cc
@@ -54,16 +54,14 @@ ExtensionFunction::ResponseAction ResourcesPrivateGetStringsFunction::Run() {
   switch (component) {
     case api::resources_private::COMPONENT_IDENTITY:
       AddStringsForIdentity(dict.get());
       break;
-#if BUILDFLAG(ENABLE_PDF)
     case api::resources_private::COMPONENT_PDF: {
 #if BUILDFLAG(ENABLE_PDF)
       pdf_extension_util::AddStrings(pdf_extension_util::PdfViewerContext::kAll,
                                      dict.get());
       pdf_extension_util::AddAdditionalData(dict.get());
 #endif  // BUILDFLAG(ENABLE_PDF)
     } break;
-#endif  // BUILDFLAG(ENABLE_PDF)
     case api::resources_private::COMPONENT_NONE:
       NOTREACHED();
   }
